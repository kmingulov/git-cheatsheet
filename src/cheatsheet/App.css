.header {
  padding-left: 15px !important;
  padding-right: 15px !important;
}

.mainContent {
  padding-top: 20px;
}

.git-command-column {
  width: 40%;
}

/* .git { } */

/* .git-command { } */

/* .git-remote { } */

.git-url {
  text-decoration: underline;
}

.git-branch {
  font-weight: bold;
}

/* .git-commit { } */

/* .git-ref { } */

.git-file {
  font-style: italic;
}
