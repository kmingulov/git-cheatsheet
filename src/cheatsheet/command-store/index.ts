export * from './CommandStore';
export * from './lunr';
