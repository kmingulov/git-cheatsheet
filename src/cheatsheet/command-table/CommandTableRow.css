/*
 * Command styles.
 */

.command-row td:nth-of-type(1) {
  width: 40%;
}

.command-content {
  padding-top: 90px;
  margin-top: -90px;
}

/*
 * Command description styles.
 */

.command-description {
  display: flex;
}

.command-anchor-link {
  visibility: hidden;
  margin-left: auto;
  font-size: larger;
}

.command-row:hover .command-anchor-link {
  visibility: visible;
}

.command-anchor-link a:hover {
  text-decoration: none;
}

/*
 * Command parts styles.
 */

.command-part-git, .command-part-command {
  color: #999;
}

.command-part-remote {
  color: #00bc8c;
}

.command-part-url {
  text-decoration: underline;
}

.command-part-ref {
  font-weight: bold;
  color: #3498db;
}

.command-part-file {
  font-style: italic;
}

.command-part-string {
  font-style: italic;
  color: #00bc8c;
}

.command-part-highlight {
  padding: 0 5px;
  background-color: white;
  color: black;
}
