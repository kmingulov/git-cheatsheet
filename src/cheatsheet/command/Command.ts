import { CommandPart } from './CommandPart';

export class Command {
  public readonly scmBreezeShortcut: Array<CommandPart | string>;
  public readonly command: Array<CommandPart | string>;
  public readonly description: Array<CommandPart | string>;

  constructor(scmBreezeShortcut: Array<CommandPart | string>,
              commandParts: Array<CommandPart | string>,
              descriptionParts: Array<CommandPart | string>) {
    this.scmBreezeShortcut = scmBreezeShortcut;
    this.command = commandParts;
    this.description = descriptionParts;
  }
}
