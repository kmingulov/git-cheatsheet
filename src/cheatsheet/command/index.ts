export * from './Command';
export * from './CommandGroup';
export * from './CommandPart';
export * from './CommandPartType';
