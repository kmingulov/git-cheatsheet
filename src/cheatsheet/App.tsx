import React, { Component, ReactElement } from 'react';
import { Container, FormControl, InputGroup, Navbar } from 'react-bootstrap';
import { FaSearch } from 'react-icons/fa';

import { CommandStore } from './command-store';
import { CommandTable } from './command-table';

import commands from './command-data/commands';

import './App.css';

const store = new CommandStore(commands);

const initialState = { searchTerm: '' };
type State = Readonly<typeof initialState>;

class App extends Component<object, State> {
  public readonly state: State = initialState;

  public render(): ReactElement {
    return (
      <div>
        <Navbar bg='primary' sticky='top'>
          <Container className='header'>
            <Navbar.Brand>Git Cheatsheet</Navbar.Brand>
          </Container>
        </Navbar>

        <Container className='mainContent'>
          <InputGroup className='searchBox'>
            <InputGroup.Prepend>
              <InputGroup.Text>
                <FaSearch/>
              </InputGroup.Text>
            </InputGroup.Prepend>
              <FormControl
                placeholder='Searchâ€¦'
                onChange={ (event: any) => this.setState({ searchTerm: event.target.value }) }
              />
          </InputGroup>

          <CommandTable store={ store } searchTerm={ this.state.searchTerm }/>
        </Container>
      </div>
    );
  }
}

export default App;
