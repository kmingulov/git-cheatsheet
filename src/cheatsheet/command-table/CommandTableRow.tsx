import React, { Component, ReactElement } from 'react';

import { Command } from '../command/Command';
import { CommandPart } from '../command/CommandPart';

const renderPart = (part: CommandPart | string, insertSpaces: boolean, key: any): ReactElement | string => {
  if (typeof(part) === 'string') {
    return (insertSpaces ? ' ' : '') + part;
  }

  const className: string = 'git-' + part.role.toString().toLocaleLowerCase();

  if (!insertSpaces) {
    return (
      <span className={ className } key={ key }>
        { part.text }
      </span>
    );
  }

  return (
    <span key={ key }>
      { ' ' }
      <span className={ className }>
        { part.text }
      </span>
    </span>
  );
};

const renderCommand = (command: Command): ReactElement => {
  const renderedCommand = (
    <span>
      <span className='git'>git</span>
      { command.commandParts.map((part, i) => renderPart(part, true, i)) }
    </span>
  );

  if (command.scmBreezeShortcut.length === 0) {
    return renderedCommand;
  }

  return (
    <span>
      { renderedCommand }
      <br/>
      <small>
        { command.scmBreezeShortcut.map((part, i) => renderPart(part, true, i)) }
      </small>
    </span>
  );
};

interface CommandTableRowProps {
  command: Command;
}

export class CommandTableRow extends Component<CommandTableRowProps> {
  public render(): ReactElement {
    const { command } = this.props;

    return (
      <tr>
        <td className='git-command-column'>
          { renderCommand(command) }
        </td>
        <td>
          { command.descriptionParts.map((part, i) => renderPart(part, false, i)) }
        </td>
      </tr>
    );
  }
}
