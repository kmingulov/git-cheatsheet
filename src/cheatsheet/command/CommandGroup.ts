import { Command } from './Command';

/**
 * Class representing a semantically bound group of commands.
 */
export class CommandGroup {
  /** Group title. */
  public readonly title: string;
  /** Array of commands. */
  public readonly commands: ReadonlyArray<Command>;

  /**
   * Constructs a new {@link CommandGroup}.
   * @param title group title
   * @param commands array of commands
   */
  constructor(title: string, commands: ReadonlyArray<Command>) {
    this.title = title;
    this.commands = commands;
  }
}
