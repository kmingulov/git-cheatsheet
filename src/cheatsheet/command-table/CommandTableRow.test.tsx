import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React, { ReactElement } from 'react';

import { Command, command, none } from 'cheatsheet/command';

import { CommandTableRow } from './CommandTableRow';

Enzyme.configure({adapter: new Adapter()});

const cmdWithShortcut = new Command(
  'cmd-1',
  [ command('commit'), none(' '), none('-m'), none('message') ],
  [ none('Commit changes in the staging area with the given message.') ],
  [ command('gc'), none(' '), none('-m'), none('message') ],
);

const cmdWithoutShortcut = new Command(
  'cmd-1',
  [ command('commit'), none(' '), none('-m'), none('message') ],
  [ none('Commit changes in the staging area with the given message.') ],
);

const renderRow = (cmd: Command): ReactElement => {
  return (
    <table>
      <tbody>
        <CommandTableRow command={ cmd }/>
      </tbody>
    </table>
  );
};

describe(CommandTableRow, () => {
  it('renders correctly a command with shortcut', () => {
    const component = mount(renderRow(cmdWithShortcut));
    expect(component).toMatchSnapshot();
  });

  it('renders correctly a command without shortcut', () => {
    const component = mount(renderRow(cmdWithoutShortcut));
    expect(component).toMatchSnapshot();
  });
});
