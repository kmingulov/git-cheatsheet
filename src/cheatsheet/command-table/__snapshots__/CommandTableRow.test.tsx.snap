// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandTableRow renders correctly a command with shortcut 1`] = `
<table>
  <tbody>
    <CommandTableRow
      command={
        Command {
          "command": Array [
            CommandPart {
              "role": "COMMAND",
              "text": "commit",
            },
            CommandPart {
              "role": "NONE",
              "text": " ",
            },
            CommandPart {
              "role": "NONE",
              "text": "-m",
            },
            CommandPart {
              "role": "NONE",
              "text": "message",
            },
          ],
          "description": Array [
            CommandPart {
              "role": "NONE",
              "text": "Commit changes in the staging area with the given message.",
            },
          ],
          "id": "cmd-1",
          "scmBreezeShortcut": Array [
            CommandPart {
              "role": "COMMAND",
              "text": "gc",
            },
            CommandPart {
              "role": "NONE",
              "text": " ",
            },
            CommandPart {
              "role": "NONE",
              "text": "-m",
            },
            CommandPart {
              "role": "NONE",
              "text": "message",
            },
          ],
        }
      }
    >
      <tr>
        <td
          className="git-command-column"
        >
          <span>
            <span>
              <span
                className="git"
              >
                git
              </span>
               
              <span
                className="git-command"
                key="0"
              >
                commit
              </span>
               
              -m
              message
            </span>
            <br />
            <small>
              <span
                className="git-command"
                key="0"
              >
                gc
              </span>
               
              -m
              message
            </small>
          </span>
        </td>
        <td>
          Commit changes in the staging area with the given message.
        </td>
      </tr>
    </CommandTableRow>
  </tbody>
</table>
`;

exports[`CommandTableRow renders correctly a command without shortcut 1`] = `
<table>
  <tbody>
    <CommandTableRow
      command={
        Command {
          "command": Array [
            CommandPart {
              "role": "COMMAND",
              "text": "commit",
            },
            CommandPart {
              "role": "NONE",
              "text": " ",
            },
            CommandPart {
              "role": "NONE",
              "text": "-m",
            },
            CommandPart {
              "role": "NONE",
              "text": "message",
            },
          ],
          "description": Array [
            CommandPart {
              "role": "NONE",
              "text": "Commit changes in the staging area with the given message.",
            },
          ],
          "id": "cmd-1",
          "scmBreezeShortcut": Array [],
        }
      }
    >
      <tr>
        <td
          className="git-command-column"
        >
          <span>
            <span
              className="git"
            >
              git
            </span>
             
            <span
              className="git-command"
              key="0"
            >
              commit
            </span>
             
            -m
            message
          </span>
        </td>
        <td>
          Commit changes in the staging area with the given message.
        </td>
      </tr>
    </CommandTableRow>
  </tbody>
</table>
`;
